README.md

```diff
@@ -10,22 +10,106 @@
+    $ cd manageiq-providers-redfish
+
+After this is done, we must perform the initial setup by running
+
+    $ bin/setup
+
+After the initial setup is done, we can test the environment by executing
+
+    $ bundle exec rake
+
+This command should run test suite and return with no error. If this is not
+the case, congratulations to us, since we can start debugging our setup;) If
+everything is green, we are ready start adding ~~bugs~~ features to the
+Redfish provider.
+
+To be able to view and test the changes we are making to the provider, we must
+override the gem sources in `~/miq/manageiq/bundler.d/overrides.rb`. In our
+case, `overrides.rb` should look something like this:
+
+    override_gem "manageiq-providers-redfish",
+      :path => File.expand_path("~/miq/manageiq-providers-redfish")
+
+Do not use relative path in `File.expand_path` call, since this is just
+calling for troubles. To actually install the local gems, run
+
+    $ ( cd ../manageiq && bin/update )
+
+Now ManageIQ will use the local manageiq-providers-redfish repository instead
+of fetching from GitHub. This will allow you to test changes and debug before
+your changes have been merged.
+
+But it will not take long for us to get into situation where we will need to
+modify some other parts of the ManageIQ in order to get new feature in. In
+order to be able to run automated test suite in such situations, we need to
+remove `~/miq/manageiq-providers-redfish/spec/manageiq` folder and replace it
+with symbolic link to `~/miq/manageiq`. To get things updated, we need to run
+
+    $ bin/update
+
+Last thing we need in order to be able to develop Redfish provider is mock
+Redfish server. And fortunately for us, it has been already installed as a
+part of an initial setup. We just need to get our hands on some recordings and
+we are good to go.
+
+One recording is available in [XLAB's repo][redfish-recordings]. We can clone
+it and instruct mock server to serve it:
+
+    $ ( cd ~/miq && git clone https://github.com/xlab-si/redfish-recordings )
+    $ bundle exec redfish serve ../redfish-recordings/lenovo-sr650
+
+   [redfish-recordings]: https://github.com/xlab-si/redfish-recordings
+                         (XLAB's repo with Redfish recordings)
+
+
+### Advanced setup (if we need changes in core)
+
+ 1. remove `spec/manageiq` and replace it with a symlink to ~/miq/manageiq
+ 2. checkout proper branch in ~/miq/manageiq
+ 3. bin/update in ~/miq/manageiq
+ 4. bundle exec rake
+
+### Testing changes manually
+
+ 1. edit ~/miq/manageiq/bundler.d/overrides.rb
+ 2. bin/update
+ 3. MIQ_SPARTAN=minimal:ems_inventory bundle exec rake evm:start
+ 4. go to localhost:3000 (admin/smartvm)
 
-For quick local setup run `bin/setup`, which will clone the core ManageIQ repository under the *spec* directory and setup necessary config files. If you have already cloned it, you can run `bin/update` to bring the core ManageIQ code up to date.
```

Comment:
I now understand that there already was this short “guide” that you now extended with more specific steps.